File: Neha Niharika Kar/Python Code/LED.py
from environment import *
from physical import *
from gpio import *
from time import *
from pyjs import *

MAX_LIGHT_PERCENT = 1
VOLUME_AT_RATE = 100000
value = 0

def setup():
    setComponentOpacity("black", 1)
    add_event_detect(0, isr)
    isr()


def isr():
    global value
    analoginput = analogRead(0)
    value = js_map(analoginput, 0, 1023, 0, 1)
    setComponentOpacity("black", 1-value)
    setDeviceProperty(getName(), "level",analoginput)



def main():
    setup()
    while True:
        updateEnvironment()
        delay(1000)


def updateEnvironment():
    global value
    global MAX_LIGHT_PERCENT
    global VOLUME_AT_RATE
    rate = float(value*MAX_LIGHT_PERCENT*VOLUME_AT_RATE) / Environment.getVolume()
    # rate equals limit because we want it to happen immediately
    Environment.setContribution("Visible Light", rate, rate, False)

if __name__ == "__main__":
    main()

File: Neha Niharika Kar/JavaScript Code/LED.js
var MAX_LIGHT_PERCENT = 1;
var VOLUME_AT_RATE = 100000;
var input;
var value;

function setup() {
	setComponentOpacity("black", 1);
	attachInterrupt(0, isr);
	isr();
}

function isr() {
	input = analogRead(0);
	value = map(input, 0, 1023, 0, 1);
	
	setComponentOpacity("black", 1-value);
	setDeviceProperty(getName(), "level",input);
}

var lastTimeInSeconds = 0;
function loop()
{
	updateEnvironment();
	
	delay(1000);
}

function updateEnvironment()
{
	var rate = value*MAX_LIGHT_PERCENT*VOLUME_AT_RATE / Environment.getVolume();
	// rate equals limit because we want it to happen immediately
	Environment.setContribution("Visible Light", rate, rate, false);
}
